spec/fsm_redirect_helper.rb

```diff
@@ -0,0 +1,23 @@
+# Conditionally monkey-patches ApplicationController#redirect_to and calls super
+# only in tests if environment variable PRINT_REDIRECTS is set.
+# To enable: $ export PRINT_REDIRECTS=true
+# $ bundle exec rspec spec/...
+# Output looks like:
+# Redirected FROM .../app/services/flow/flow_state_machine.rb:170:in `redirect_to_step'
+#   TO: "http://127.0.0.1:50654/verify/doc_auth/link_sent"
+# To disable: $ unset PRINT_REDIRECTS
+# Handy for debugging the Flow State Machine.
+class ApplicationController
+  if ENV['PRINT_REDIRECTS']
+    def redirect_to(options = {}, response_status = {})
+      # rubocop:disable Rails/Output
+      # rubocop:disable Style/RescueModifier
+      print("Redirected FROM #{caller(1..1).first rescue "unknown"}\n")
+      print("  TO: #{options.inspect}\n")
```

Comment:
I would use `puts` so that the newlines are automatically added
```suggestion
      puts "Redirected FROM #{caller(1..1).first rescue "unknown"}"
      print "TO: #{options.inspect}"
```